gensim/models/tfidfmodel.py

```diff
@@ -144,17 +262,27 @@ def __getitem__(self, bow, eps=1e-12):
 
         # unknown (new) terms will be given zero weight (NOT infinity/huge weight,
         # as strict application of the IDF formula would dictate)
+
+        termid_array, tf_array = [], []
+        for termid, tf in bow:
+            termid_array.append(termid)
+            tf_array.append(tf)
+
+        tf_array = self.wlocal(np.array(tf_array))
+
         vector = [
-            (termid, self.wlocal(tf) * self.idfs.get(termid))
-            for termid, tf in bow if self.idfs.get(termid, 0.0) != 0.0
+            (termid, tf * self.idfs.get(termid))
+            for termid, tf in zip(termid_array, tf_array) if self.idfs.get(termid, 0.0) != 0.0
```

Comment:
This is too brittle; compare floats for equality using `eps` instead: For our purposes, two floats are "equal" if their absolute difference is smaller than `eps`.

Reply:
I'm confused, https://github.com/RaRe-Technologies/gensim/pull/1791#discussion_r160090398 and https://github.com/RaRe-Technologies/gensim/pull/1791#discussion_r158868854 contradict each other, what you mean @piskvorky?

Reply:
To compare float for equality, use `abs(x - y) < eps`. Not `x == y`, nor `x - y < eps`.